Lib/test/test_logging.py

```diff
@@ -4086,6 +4086,13 @@ def test_invalid_names(self):
         self.assertRaises(TypeError, logging.getLogger, any)
         self.assertRaises(TypeError, logging.getLogger, b'foo')
 
+    def test_pickling(self):
```

Comment:
`__reduce__` is used also in the `copy` module. Maybe add tests for `copy.copy()` and `copy.deepcopy()`? Even unpickleable loggers can be copied if add special `__copy__` and `__deepcopy__` methods or register the classes with `copyreg.pickle()`.